### PR TITLE
[tune] Annotated Example Page and showcase Tutorials

### LINKED ISSUE #2728: [doc][tune] Annotated Examples + Link to Tutorial
### Describe the problem
It would be great to annotate a couple key examples (i.e., Keras, PyTorch) for Tune on the documentation page and also some notes on the source code README.

A FAQ would be nice, now that more issues have been raised.

### PR DESCRIPTION
**Pull request by richardliaw**
Adds an example page and link in codebase.

Closes #2728.

### CODE REVIEW

File: python/ray/tune/examples/README.rst [hunk 1]
```diff
@@ -1,4 +1,60 @@
 Tune Examples
 =============
 
-Code examples for various schedulers and Tune features.
+.. Keep this in sync with ray/doc/tune-examples.rst
```
Comment (ericl) [inline python/ray/tune/examples/README.rst]: Could we just link to the readthedocs here? Is there an advantage of having a README that's buried in the repo?

**Review by ericl — APPROVED** [review]

File: python/ray/tune/examples/README.rst (hunk 1 above)
Comment (richardliaw) [reply python/ray/tune/examples/README.rst]: Well the examples here are intended to be actively viewed and perused, so this is a natural landing page IMO.
